service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/models/DigitalTwinCommandResponse.java

```diff
@@ -27,12 +30,22 @@ public void setStatus(Integer status)
         this.status = status;
     }
 
-    public String getPayload()
+    public JsonElement getPayloadAsJsonElement()
     {
         return payload;
     }
 
-    public void setPayload(String payload)
+    public String getPayloadAsJsonString()
+    {
+        return getPayloadAsCustomType(String.class);
+    }
+
+    public <T> T getPayloadAsCustomType(Class<T> customObject)
+    {
+        return new GsonBuilder().create().fromJson(payload, customObject);
+    }
+
+    public void setPayload(JsonElement payload)
```

Comment:
Users won't be setting this payload, right? If so, then you can remove this setter or make it package private

Reply:
this methods is called by `Tools.java` in different packages. See this [line](https://github.com/Azure/azure-iot-sdk-java/blob/1a3b02bd10928ddebccc9a1fee77f23eab3e41dc/service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/helpers/Tools.java#L42).